services/charts_updater.go

```diff
@@ -1467,24 +1469,174 @@ func depositsDistributionChartData() (*types.GenericChartData, error) {
-		Y    uint64 `json:"y"`
+		Name      string `json:"name"`
+		Address   string `json:"address"`
+		Y         uint64 `json:"y"`
+		Drilldown string `json:"drilldown"`
+	}
+
+	seriesData := []seriesDataItem{
+		{
+			Name:      "Kraken",
+			Y:         0,
+			Drilldown: "Kraken",
+		},
+		{
+			Name:      "Binance",
+			Y:         0,
+			Drilldown: "Binance",
+		},
+		{
+			Name:      "Whale",
+			Y:         0,
+			Drilldown: "Whale",
+		},
+		{
+			Name:      "Huobi",
+			Y:         0,
+			Drilldown: "Huobi",
+		},
+		{
+			Name:      "Bitcoin suisse",
+			Y:         0,
+			Drilldown: "Bitcoin suisse",
+		},
+		{
+			Name:      "Staked.us",
+			Y:         0,
+			Drilldown: "Staked.us",
+		},
+		{
+			Name:      "Lido",
+			Y:         0,
+			Drilldown: "Lido",
+		},
+		{
+			Name:      "Stakefish",
+			Y:         0,
+			Drilldown: "Stakefish",
+		},
+		{
+			Name:      "Defi",
+			Y:         0,
+			Drilldown: "Defi",
+		},
+	}
+	drillSeries := []drillSeriesData{
+		{
+			Name: "Kraken",
+			ID:   "Kraken",
+			Data: [][2]string{},
+		},
+		{
+			Name: "Binance",
+			ID:   "Binance",
+			Data: [][2]string{},
+		},
+		{
+			Name: "Whale",
+			ID:   "Whale",
+			Data: [][2]string{},
+		},
+		{
+			Name: "Huobi",
+			ID:   "Huobi",
+			Data: [][2]string{},
+		},
+		{
+			Name: "Bitcoin suisse",
+			ID:   "Bitcoin suisse",
+			Data: [][2]string{},
+		},
+		{
+			Name: "Staked.us",
+			ID:   "Staked.us",
+			Data: [][2]string{},
+		},
+		{
+			Name: "Lido",
+			ID:   "Lido",
+			Data: [][2]string{},
+		},
+		{
+			Name: "Stakefish",
+			ID:   "Stakefish",
+			Data: [][2]string{},
+		},
+		{
+			Name: "Defi",
+			ID:   "Defi",
+			Data: [][2]string{},
+		},
+		{ // always must be the last
+			Name: "Others",
+			ID:   "Others",
+			Data: [][2]string{},
+		},
 	}
-	seriesData := []seriesDataItem{}
 	othersItem := seriesDataItem{
-		Name: "others",
-		Y:    0,
+		Name:      "Others",
+		Y:         0,
+		Drilldown: "Others",
 	}
 	for i := range rows {
-		if i > 20 {
-			othersItem.Y += rows[i].Count
+		var poolName string = "Unknown"
+		curAddr := string(utils.FormatEth1AddressString(rows[i].Address))
+		skipAppend := false
+		for _, pool := range stakePools {
+			if strings.ToLower(curAddr) == strings.ToLower("0x"+pool.Address) {
+				for j, seriesItem := range seriesData {
+					if strings.Contains(pool.Name, seriesItem.Name) {
+						seriesData[j].Y += rows[i].Count
+						for k, drillItem := range drillSeries {
+							if strings.Contains(pool.Name, drillItem.ID) {
+								drillSeries[k].Data = append(drillSeries[k].Data, [2]string{pool.Name, fmt.Sprintf("%d", rows[i].Count)})
+								break
+							}
+						}
+						skipAppend = true
+						break
+					}
+				}
+				poolName = pool.Name
+				break
+			}
+		}
+
+		if skipAppend {
 			continue
 		}
-		seriesData = append(seriesData, seriesDataItem{
-			Name: string(utils.FormatEth1AddressString(rows[i].Address)),
-			Y:    rows[i].Count,
-		})
+
+		if drillSeries[len(drillSeries)-1].ID == "Others" {
```

Comment:
if len(drillSeries) is 0 this will cause a runtime exception

Reply:
The drillSeries  is hard-coded so it always has a length grater than one, just have to make sure that the last one is always "Others"